yarn.lock

```diff
@@ -1009,6 +1009,34 @@
     "@types/istanbul-reports" "^1.1.1"
     "@types/yargs" "^13.0.0"
 
+"@nodelib/fs.scandir@2.1.3":
+  version "2.1.3"
+  resolved "https://registry.yarnpkg.com/@nodelib/fs.scandir/-/fs.scandir-2.1.3.tgz#3a582bdb53804c6ba6d146579c46e52130cf4a3b"
+  integrity sha512-eGmwYQn3gxo4r7jdQnkrrN6bY478C3P+a/y72IJukF8LjB6ZHeB3c+Ehacj3sYeSmUXGlnA67/PmbM9CVwL7Dw==
+  dependencies:
+    "@nodelib/fs.stat" "2.0.3"
+    run-parallel "^1.1.9"
+
+"@nodelib/fs.stat@2.0.3", "@nodelib/fs.stat@^2.0.2":
+  version "2.0.3"
+  resolved "https://registry.yarnpkg.com/@nodelib/fs.stat/-/fs.stat-2.0.3.tgz#34dc5f4cabbc720f4e60f75a747e7ecd6c175bd3"
+  integrity sha512-bQBFruR2TAwoevBEd/NWMoAAtNGzTRgdrqnYCc7dhzfoNvqPzLyqlEQnzZ3kVnNrSp25iyxE00/3h2fqGAGArA==
```

Comment:
@shcheklein 

> looks like too many changes just to this file for a single library.

I tested some more new ones. Summary:

```console
$ yarn why @nodelib/fs.stat
   - "lint-staged#del#globby#fast-glob" depends on it
$ yarn why @nodelib/fs.walk
   - "lint-staged#del#globby#fast-glob" depends on it
$ yarn why @samverschueren/stream-to-observable
   - "lint-staged#listr" depends on it
$ yarn why @types/events
   - "lint-staged#del#globby#@types#glob" depends on it
$ yarn why @types/glob
   - "lint-staged#del#globby" depends on it
$ yarn why @types/minimatch
   - "lint-staged#del#globby#@types#glob" depends on it
$ yarn why @types/node
   - "lint-staged#del#globby#@types#glob" depends on it
$ yarn why aggregate-error
   - "lint-staged#del#p-map" depends on it
$ yarn why any-observable
   - "lint-staged#listr#@samverschueren#stream-to-observable" depends on it
$ yarn why array-union
   - Hoisted from "pretty-quick#multimatch#array-union"
```

Only at `array-union` did the reason stop being `lint-staged`, but it comes from `pretty-quick` instead, which we also use. (Not sure why it wasn't there before.)

There's many other new ones below it seems, but I haven't tried them all... Will need more time to finish investigating this.

Reply:
OK I tested down to `path-key` and actually it seems indeed everything new is somehow related to `lint-staged`. Sometimes they're also related to other previously installed packages so maybe this file just hasn't been updated in the repo for a long time. But in any case `lint-staged` does have a ridiculous amount of dependencies...

<details>

<summary>Summary of yarn.lock package reasons (`array-union` to `p-map`)</summary>

```console
$ yarn why array-union
=> Found "array-union@1.0.2"
   - Hoisted from "pretty-quick#multimatch#array-union"
   - Hoisted from "next#autodll-webpack-plugin#del#globby#array-union"
=> Found "globby#array-union@2.1.0"
info This module exists because "lint-staged#del#globby" depends on it.
$ yarn why braces
=> Found "braces@2.3.2"
   - Hoisted from "micromatch#braces"
   - Hoisted from "next#fork-ts-checker-webpack-plugin#chokidar#braces"
=> Found "lint-staged#braces@3.0.2"
   - "lint-staged#micromatch" depends on it
=> Found "fast-glob#braces@3.0.2"
   - "lint-staged#del#globby#fast-glob#micromatch" depends on it
$ yarn why clean-stack
   - "lint-staged#del#p-map#aggregate-error" depends on it
$ yarn why cli-truncate
   - "lint-staged#listr#listr-update-renderer" depends on it
$ yarn why cross-spawn
=> Found "cross-spawn@6.0.5"
   - Hoisted from "execa#cross-spawn"
   - Hoisted from "eslint#cross-spawn"
=> Found "pretty-quick#cross-spawn@5.1.0"
   - "pretty-quick#execa" depends on it
=> Found "lint-staged#cross-spawn@7.0.1"
   - "lint-staged#execa" depends on it
$ yarn why date-fns
   - "lint-staged#listr#listr-verbose-renderer" depends on it
$ yarn why dedent
   - "lint-staged" depends on it
$ yarn why del
=> Found "del@5.1.0"
   - "lint-staged" depends on it
=> Found "autodll-webpack-plugin#del@3.0.0"
info This module exists because "next#autodll-webpack-plugin" depends on it.
$ yarn why dir-glob
   - "lint-staged#del#globby" depends on it
$ yarn why elegant-spinner
   - "lint-staged#listr#listr-update-renderer" depends on it
$ yarn why execa
=> Found "execa@1.0.0"
   - Hoisted from "husky#execa"
   - Hoisted from "jest#jest-cli#@jest#core#jest-changed-files#execa"
   - Hoisted from "babel-jest#@jest#transform#jest-haste-map#sane#execa"
=> Found "pretty-quick#execa@0.8.0"
info This module exists because "pretty-quick" depends on it.
=> Found "lint-staged#execa@2.1.0"
info This module exists because "lint-staged" depends on it.
$ yarn why fast-glob
   - "lint-staged#del#globby" depends on it
$ yarn why fastq
   - "lint-staged#del#globby#fast-glob#@nodelib#fs.walk" depends on it
$ yarn why figures
=> Found "figures@2.0.0"
   - Hoisted from "eslint#inquirer#figures"
   - Hoisted from "lint-staged#listr#listr-verbose-renderer#figures"
   - Hoisted from "styled-reset#opencollective#inquirer#figures"
=> Found "listr-update-renderer#figures@1.7.0"
info This module exists because "lint-staged#listr#listr-update-renderer" depends on it.
$ yarn why fill-range
=> Found "fill-range@4.0.0"
   - "braces" depends on it
   - Hoisted from "braces#fill-range"
=> Found "lint-staged#fill-range@7.0.1"
   - "lint-staged#micromatch#braces" depends on it
=> Found "fast-glob#fill-range@7.0.1"
   - "lint-staged#del#globby#fast-glob#micromatch#braces" depends on it
$ yarn why get-own-enumerable-property-symbols
   - "lint-staged#stringify-object" depends on it
$ yarn why get-stream
=> Found "get-stream@4.1.0"
   - "execa" depends on it
=> Found "pretty-quick#get-stream@3.0.0"
   - "pretty-quick#execa" depends on it
=> Found "lint-staged#get-stream@5.1.0"
   - "lint-staged#execa" depends on it
$ yarn why globby
=> Found "globby@10.0.1"
   - "lint-staged#del" depends on it
=> Found "autodll-webpack-plugin#globby@6.1.0"
   - "next#autodll-webpack-plugin#del" depends on it
$ yarn why graceful-fs
=> Found "graceful-fs@4.2.2"
   - Hoisted from "babel-jest#@jest#transform#graceful-fs"
   - Hoisted from ...
   - Hoisted from "next#webpack#terser-webpack-plugin#cacache#move-concurrently#fs-write-stream-atomic#graceful-fs"
   - Hoisted from "babel-jest#babel-plugin-istanbul#test-exclude#read-pkg-up#read-pkg#load-json-file#graceful-fs"
=> Found "del#graceful-fs@4.2.3"
info This module exists because "lint-staged#del" depends on it.
$ yarn why ignore
=> Found "ignore@3.3.10"
   - "pretty-quick" depends on it
   - Hoisted from "pretty-quick#ignore"
=> Found "eslint#ignore@4.0.6"
info This module exists because "eslint" depends on it.
=> Found "globby#ignore@5.1.4"
info This module exists because "lint-staged#del#globby" depends on it.
$ yarn why indent-string
=> Found "indent-string@3.2.0"
   - "lint-staged#listr#listr-update-renderer" depends on it
=> Found "aggregate-error#indent-string@4.0.0"
info This module exists because "lint-staged#del#p-map#aggregate-error" depends on it.
$ yarn why is-number
=> Found "is-number@3.0.0"
   - Hoisted from "braces#fill-range#is-number"
   - Hoisted from "braces#fill-range#to-regex-range#is-number"
   - Hoisted from "braces#snapdragon#base#cache-base#has-value#has-values#is-number"
=> Found "lint-staged#is-number@7.0.0"
   - "lint-staged#micromatch#braces#fill-range#to-regex-range" depends on it
=> Found "fast-glob#is-number@7.0.0"
   - "lint-staged#del#globby#fast-glob#micromatch#braces#fill-range#to-regex-range" depends on it
$ yarn why is-obj
   - "lint-staged#stringify-object" depends on it
$ yarn why is-observable
   - "lint-staged#listr" depends on it
$ yarn why is-path-cwd
=> Found "is-path-cwd@2.2.0"
   - "lint-staged#del" depends on it
=> Found "autodll-webpack-plugin#is-path-cwd@1.0.0"
   - "next#autodll-webpack-plugin#del" depends on it
$ yarn why is-path-inside
=> Found "is-path-inside@3.0.2"
   - "lint-staged#del" depends on it
=> Found "is-path-in-cwd#is-path-inside@1.0.1"
info This module exists because "next#autodll-webpack-plugin#del#is-path-in-cwd" depends on it.
$ yarn why is-regexp
   - "lint-staged#stringify-object" depends on it
$ yarn why is-stream
=> Found "is-stream@1.1.0"
   - Hoisted from "execa#is-stream"
   - Hoisted from ...
   - Hoisted from "styled-reset#opencollective#node-fetch#is-stream"
=> Found "lint-staged#is-stream@2.0.0"
   - "lint-staged#execa" depends on it
$ yarn why lint-staged
info Has been hoisted to "lint-staged"
$ yarn why listr...
   - "lint-staged..." depends on it
$ yarn why log-symbols
=> Found "log-symbols@3.0.0"
   - "lint-staged" depends on it
=> Found "listr-update-renderer#log-symbols@1.0.2"
info This module exists because "lint-staged#listr#listr-update-renderer" depends on it.
$ yarn why merge2
   - "lint-staged#del#globby" depends on it
$ yarn why micromatch
=> Found "micromatch@3.1.10"
info Has been hoisted to "micromatch"
info Reasons this module exists
   - Hoisted from "next#webpack#micromatch"
   - Hoisted from ...
   - Hoisted from "babel-jest#@jest#transform#jest-haste-map#sane#micromatch"
=> Found "lint-staged#micromatch@4.0.2"
info This module exists because "lint-staged" depends on it.
=> Found "fast-glob#micromatch@4.0.2"
info This module exists because "lint-staged#del#globby#fast-glob" depends on it.
$ yarn why mimic-fn
=> Found "mimic-fn@2.1.0"
   - "lint-staged#execa#onetime" depends on it
=> Found "restore-cursor#mimic-fn@1.2.0"
   - "eslint#inquirer#cli-cursor#restore-cursor#onetime" depends on it
$ yarn why npm-run-path
=> Found "npm-run-path@2.0.2"
   - Hoisted from "execa#npm-run-path"
   - Hoisted from "pretty-quick#execa#npm-run-path"
=> Found "lint-staged#npm-run-path@3.1.0"
   - "lint-staged#execa" depends on it
$ yarn why onetime
info Reasons this module exists
   - "lint-staged#execa" depends on it
=> Found "restore-cursor#onetime@2.0.1"
info This module exists because "eslint#inquirer#cli-cursor#restore-cursor" depends on it.
$ yarn why p-finally
=> Found "p-finally@1.0.0"
   - Hoisted from "execa#p-finally"
   - Hoisted from "pretty-quick#execa#p-finally"
=> Found "lint-staged#p-finally@2.0.1"
   - "lint-staged#execa" depends on it
$ yarn why p-map
=> Found "p-map@2.1.0"
   - "lint-staged#listr" depends on it
=> Found "del#p-map@3.0.0"
info This module exists because "lint-staged#del" depends on it.
=> Found "autodll-webpack-plugin#p-map@1.2.0"
   - "next#autodll-webpack-plugin#del" depends on it
$ yarn why p-map
=> Found "p-map@2.1.0"
   - "lint-staged#listr" depends on it
=> Found "del#p-map@3.0.0"
info This module exists because "lint-staged#del" depends on it.
=> Found "autodll-webpack-plugin#p-map@1.2.0"
   - "next#autodll-webpack-plugin#del" depends on it
```

</details>

Reply:
👍